src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -1,25 +1,171 @@
-	{
-	    request.getRequestDispatcher("/WEB-INF/view/activityfeed.jsp").forward(request, response);
-	}
-}
+package codeu.controller;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.*;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+
+/** Servlet class responsible for the activity feed page. */
+public class ActivityFeedServlet extends HttpServlet {
+	/** Store class that gives access to Conversations. */
+	private ConversationStore conversationStore;
+
+	/** Store class that gives access to Messages. */
+	private MessageStore messageStore;
+
+	/** Store class that gives access to Users. */
+	private UserStore userStore;
+
+	ArrayList<Instant> arrInstantsSorted = new ArrayList<Instant>();
+
+	/** Set up state for handling chat requests. */
+	@Override
+	public void init() throws ServletException {
+		super.init();
+		setConversationStore(ConversationStore.getInstance());
+		setMessageStore(MessageStore.getInstance());
+		setUserStore(UserStore.getInstance());
+	}
+
+	/**
+	 * Sets the ConversationStore used by this servlet. This function provides a
+	 * common setup method for use by the test framework or the servlet's init()
+	 * function.
+	 */
+	void setConversationStore(ConversationStore conversationStore) {
+		this.conversationStore = conversationStore;
+	}
+
+	/**
+	 * Sets the MessageStore used by this servlet. This function provides a common
+	 * setup method for use by the test framework or the servlet's init() function.
+	 */
+	void setMessageStore(MessageStore messageStore) {
+		this.messageStore = messageStore;
+	}
+
+	/**
+	 * Sets the UserStore used by this servlet. This function provides a common
+	 * setup method for use by the test framework or the servlet's init() function.
+	 */
+	void setUserStore(UserStore userStore) {
+		this.userStore = userStore;
+	}
+
+	/**
+	 * This function takes in a list of conversations,users and messages and returns
+	 * a HashMap of Instants mapping to HashMaps of UUIDs mapping to a data store:
+	 * User,Conversation or Message
+	 * 
+	 * @param conversations
+	 * @param users
+	 * @param messages
+	 * @return built HashMap
+	 */
+	HashMap<Instant, HashMap<UUID, String>> buildHashMap(List<Conversation> conversations, List<User> users,
+			List<Message> messages) {
+		HashMap<Instant, HashMap<UUID, String>> outerhm = new HashMap<Instant, HashMap<UUID, String>>();
+		if (users != null) {
+		for (User user : users) {
+			HashMap<UUID, String> innerhm = new HashMap<UUID, String>();
+			innerhm.put(user.getId(), "user");

```

Comment:
I would use strongly-typed classes over dictionaries. Also, I would suggest using trees instead of dictionaries structures, as they are inherently sorted. For example:

```
public interface Event {
  String describeEvent();
}

public class UserCreationEvent {
  UUID userId;
  UserCreationEvent(UUID u) { /* ... */ }
  String describeEvent() { /* ... */ }
}
/* ... */

void addUserCreationEvents(TreeMap<Instant, Event> events, List<User> users);
void addConversationCreationEvents(TreeMap<Instant, Event> events, List<Conversation> convos);
void addMessageEvents(TreeMap<Instant, Event> events, List<Message> message);
```

(In a production implementation, we'd probably want to use a multi-map or bag... but the Java standard library makes these a little annoying to implement, and this is probably good enough.)